jetty-server/src/main/java/org/eclipse/jetty/server/session/FileSessionDataStore.java

```diff
@@ -206,60 +207,64 @@ public void sweepDisk()
         long now = System.currentTimeMillis();
         if (LOG.isDebugEnabled())
             LOG.debug("Sweeping {} for old session files", _storeDir);
-        try
+        try (Stream<Path> stream = Files.walk(_storeDir.toPath(), 1, FileVisitOption.FOLLOW_LINKS))
         {
-            Files.walk(_storeDir.toPath(), 1, FileVisitOption.FOLLOW_LINKS)
+            stream
                 .filter(p -> !Files.isDirectory(p)).filter(p -> !isOurContextSessionFilename(p.getFileName().toString()))
                 .filter(p -> isSessionFilename(p.getFileName().toString()))
-                .forEach(p ->
-                {
-
-                    try
-                    {
-                        sweepFile(now, p);
-                    }
-                    catch (Exception e)
-                    {
-                        LOG.warn(e);
-                    }
-                });
+                .filter(p -> isExpired(now, p))
+                .forEach(this::deleteFile);
         }
         catch (Exception e)
         {
             LOG.warn(e);
         }
     }
 
+    public boolean isExpired(long now, Path p)
```

Comment:
I don't see the benefit of splitting sweepFile() into isExpired() and then deleteFile() - it's more efficient just to use sweepFile().

Reply:
Ah now that we no longer collect before delete, there is none....